articles/api-auth/tutorials/verify-access-token.md

```diff
@@ -0,0 +1,116 @@
+
+The API needs to check if the algorithm, as specified by the JWT header (property `alg`), matches the one expected by the API. If not, the token is considered invalid and the request must be rejected.
+
+In this case the mismatch might be due to mistake (it is common that the tokens are signed using the `HS256` signing algorithm, but your API is configured for `RS256`, or vice versa), but it could also be due to an attack, hence the request has to be rejected.
+
+### How can my API check the signature algorithm?
+
+To check if the signature matches the API's expectations, you have to decode the JWT and retrieve the `alg` property of the JWT header.
+
+Alternatively, you can use one of the libraries listed in the _Libraries for Token Signing/Verification_ section of [JWT.io](https://jwt.io/).
+
+Following the Node.js example of the previous section, the [jwt.verify()](https://github.com/auth0/node-jsonwebtoken#jwtverifytoken-secretorpublickey-options-callback) method of the [node-jsonwebtoken library](https://github.com/auth0/node-jsonwebtoken), supports an `algorithms` argument, that contains a list of strings with the names of the allowed algorithms.
+
+## Verify the Signature
+
+will then download all signing keys from the JWKS endpoint and see if a one of the signing keys matches the `kid` in the header of the JWT. If none of the signing keys match the incoming `kid`, an error will be thrown. If we have a match, we will pass the right signing key to express-jwt.
+
+### How can my API verify the signature?
+
+## Validate the Claims
+
+Once the API verifies the token's signature, the next step is to validate the standard claims of the token's payload. The following validations need to be made:
+
+- _Token expiration_: The current date/time _must_ be before the expiration date/time listed in the `exp` claim (which is a Unix timestamp). If not, the request must be rejected.
+
+- _Token issuer_: The `iss` claim denotes the issuer of the JWT. The value _must_ match the one configured in your API. For JWTs issued by Auth0, `iss` holds your Auth0 domain with a `https://` prefix and a `/` suffix: `https://${account.namespace}/`.
+
+- _Token audience_: The `aud` claim identifies the recipients that the JWT is intended for. For JWTs issued by Auth0, `aud` holds the unique identifier of the target API (field __Identifier__ at your [API's Settings](${manage_url}/#/apis)). If the API is not the intended audience of the JWT, it _must_ reject the request.
+
+### How can my API validate the claims?
+
+To validate the claims, you have to decode the JWT, retrieve the claims (`exp`, `iss`, `aud`) and validate their values.
+
+The easiest way however, is to use one of the libraries listed in the _Libraries for Token Signing/Verification_ section of [JWT.io](https://jwt.io/). Note that not all libraries validate all the claims. In [JWT.io](https://jwt.io/) you can see which validations each library supports (look for the green check marks).
```

Comment:
@mpaktiti It is good to mention these libraries, but note that they are typically used **only** to validate the token. 

It may also be worth mentioning that many web frameworks (such as ASP.NET Core for example) has JWT middleware which handle the token validation as well. Most of the times this will be a better route to take, rather that resorting to using one of these libraries, as the middleware typically integrates well with the framework's overall authentication mechanisms.